engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowMetadataHelper.scala

```diff
@@ -24,7 +22,7 @@ trait WorkflowMetadataHelper {
   }
   
   def pushWorkflowFailures(workflowId: WorkflowId, failures: List[Throwable]) = {
-    val failureEvents = failures flatMap { r => throwableToMetadataEvents(MetadataKey(workflowId, None, s"${WorkflowMetadataKeys.Failures}[${Random.nextInt(Int.MaxValue)}]"), r) }
+    val failureEvents = failures flatMap { r => throwableToMetadataEvents(MetadataKey(workflowId, None, s"${WorkflowMetadataKeys.Failures}"), r) }
```

Comment:
The ID is now generated in `throwableToMetadataEvents` so that it can override with non-randoms for the causes.